### PR TITLE
chore(infra): improve the configuration of `justfile`

### PR DESCRIPTION
**Pull request by shulaoda**
## Summary
Related to #7516 

- [x] support windows shell
- [x] support setup
- [x] support ready
- [x] support build
- [x] support watch

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

**Comment by hardfist** [comment]
excellent work!

File: crates/rspack_allocator/Cargo.toml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 [package]
-edition    = "2021"
-license    = "MIT"
-name       = "rspack_allocator"
-repository = "https://github.com/web-infra-dev/rspack"
-version    = "0.1.0"
 description = "rspack allocator"
+edition     = "2021"
```
Comment (hardfist) [inline crates/rspack_allocator/Cargo.toml:3]: this is changed by toml formatter?

File: justfile [hunk 2]
```diff
@@ -1,6 +1,82 @@
-# Setup the tools needed to develop
-setup-tools:
-    cargo install cargo-release insta 
-# publish rust crates    
-release-crates:
+#!/usr/bin/env -S just --justfile
```
Comment (hardfist) [inline justfile:1]: this is used for run justfile directly like `./justfile` ?
do you add execute permission for this file?

File: crates/rspack_allocator/Cargo.toml (hunk 1 above)
Comment (shulaoda) [reply crates/rspack_allocator/Cargo.toml:3]: yes

**Comment by hardfist** [comment]
@shulaoda let's separate this into multi pr, no need to finish  things in on PR, current justfile improvement can be landed first.

File: justfile [hunk 3]
```diff
@@ -1,6 +1,82 @@
...
+# Check readiness of the project
+ready:
+  just fmt  
+  just check  
+  just lint 
+  just test
+  @echo '✅ All passed!'
+
+# Publish Rust crates to crates.io
+release:
```
Comment (hardfist) [inline justfile:32]: release-crates or release-rust? since we have different ways to release javascript and rust

Comment (shulaoda) [reply justfile]: `release-rust` may be more intuitive and convenient.

**Review by hardfist — APPROVED** [review]

**Comment by hardfist** [comment]
thanks!

**Pull request merged**